File: Integration Examples/eAPI v1.6/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/EchoRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{

    public class EchoRequest : SignBase, IBaseRequest
    {
        [JsonProperty("merchantId")]
        public string MerchantId { get; set; }

        public string ToSign()
        {
            var sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, DateTime);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/EchoRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;


import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@RequiredArgsConstructor
public class EchoRequest extends SignBase {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@NonNull
	private String merchantId;

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, merchantId);
		add(sb, dttm);
		deleteLast(sb);
		return sb.toString();
	}

}
